File: Source/Aladdin.ASN1.ISO/ASN1/ISO/OCSP/RevokedInfo.cs
using System;
using System.Runtime.Serialization;

// RevokedInfo ::= SEQUENCE {
//    revocationTime                GeneralizedTime,
//    revocationReason [0] EXPLICIT CRLReason OPTIONAL
// }

namespace Aladdin.ASN1.ISO.OCSP
{
	[Serializable]
	public class RevokedInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<GeneralizedTime              >().Factory(), Cast.N                 ), 
			new ObjectInfo(new ObjectCreator<Enumerated<PKIX.CE.CrlReason>>().Factory(), Cast.EO, Tag.Context(0)) 
		}; 
		// конструктор при сериализации
        protected RevokedInfo(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public RevokedInfo(IEncodable encodable) : base(encodable, info) {} 

		// конструктор при закодировании
		public RevokedInfo(GeneralizedTime revocationTime, Enumerated<PKIX.CE.CrlReason> revocationReason) : 
			base(info, revocationTime, revocationReason) {} 

		public GeneralizedTime					RevocationTime		{ get { return (GeneralizedTime					)this[0]; } } 
		public Enumerated<PKIX.CE.CrlReason>	RevocationReason	{ get { return (Enumerated<PKIX.CE.CrlReason>   )this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/ocsp/RevokedInfo.java
package aladdin.asn1.iso.ocsp;
import aladdin.asn1.*;
import java.io.*;

// RevokedInfo ::= SEQUENCE {
//    revocationTime                GeneralizedTime,
//    revocationReason [0] EXPLICIT CRLReason OPTIONAL
// }

public class RevokedInfo extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(GeneralizedTime.class).factory(), Cast.N                 ), 
		new ObjectInfo(new ObjectCreator(Enumerated     .class).factory(), Cast.EO, Tag.context(0)), 
	}; 
	// конструктор при раскодировании
	public RevokedInfo(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public RevokedInfo(GeneralizedTime revocationTime, Enumerated revocationReason) 
	{ 
		super(info, revocationTime, revocationReason); 
	}
	public final GeneralizedTime revocationTime  () { return (GeneralizedTime)get(0); } 
	public final Enumerated      revocationReason() { return (Enumerated     )get(1); }
}
